xgi/generators/uniform.py

```diff
@@ -299,27 +370,69 @@ def uniform_erdos_renyi_hypergraph(n, m, p, p_type="degree", seed=None):
     H.add_nodes_from(node_labels)
 
     if p_type == "degree":
-        q = p / (m * n ** (m - 1))  # wiring probability
+        if multiedges:
+            q = p / (m * n ** (m - 1))  # wiring probability
+        else:
+            q = p * n / (m * comb(n, m))
     elif p_type == "prob":
         q = p
     else:
         raise XGIError("Invalid p_type!")
 
     if q > 1 or q < 0:
-        raise XGIError("Probability not in [0,1].")
+        raise XGIError("Probability not in [0, 1].")
+
+    if q == 1 and not multiedges:
+        return complete_hypergraph(n, order=m - 1)
+    if q == 0:
+        H = empty_hypergraph()
+        H.add_nodes_from(range(n))
+        return H
 
     index = np.random.geometric(q) - 1  # -1 b/c zero indexing
-    max_index = n**m
-    while index < max_index:
-        e = set(_index_to_edge(index, n, m))
+    if multiedges:
+        max_index = n**m
+        f = _index_to_edge_prod
+    else:
+        max_index = comb(n, m, exact=True)
+        f = _index_to_edge_comb
+
+    while index <= max_index:
+        e = set(f(index, n, m))
+        # if f corresponds to _index_to_edge_prod,
+        # edge ids are not guaranteed to be unique
+        # and when casting to a set, they will no
+        # longer be of size m.
+        # for instance (0, 0, 0) becomes {0}
+        # if we accept these edges, the hypergraph
+        # will not longer be uniform, so we discard them.
         if len(e) == m:
             H.add_edge(e)
         index += np.random.geometric(q)
```

Comment:
just checking, here we don't need the -1 like above because it's not the initial number right?

Reply:
Exactly. If the minimum was zero (i.e., we subtracted 1 from the right-hand side), you might sample the same index multiple times. I'll add a note.

Reply:
fixed.